### PR TITLE
Minimal tensorflow test

### PR DESCRIPTION
**Pull request by lugi0**
Signed-off-by: Luca Giorgi <lgiorgi@redhat.com>

### CODE REVIEW

**Comment by lugi0** [comment]
This PR cannot be currently tested since the spawning of the image is blocked

File: tests/Tests/500__jupyterhub/minimal-tensorflow-test.robot [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Library          DebugLibrary
+Library          JupyterLibrary
+Suite Setup      Begin Web Test
+Suite Teardown   End Web Test
+
+*** Variables ***
+
+
+*** Test Cases ***
+Minimal CUDA test
```
Comment (vasukulkarni) [inline tests/Tests/500__jupyterhub/minimal-tensorflow-test.robot:15]: regression tag?

Comment (lugi0) [reply tests/Tests/500__jupyterhub/minimal-tensorflow-test.robot]: Added in latest commit

**Comment by lugi0** [comment]
PR test:
https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/99/

**Comment by lugi0** [comment]
Fixed the import error in the PR test.
New PR test:
https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/101/

**Comment by lugi0** [comment]
Unexpected failures, rerunning: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/207/

**Comment by lugi0** [comment]
PR test here: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/210/
Last run was aborted due to issues with the user's PVC

**Comment by lugi0** [comment]
Rerunning, still issues with the PVC: https://opendatascience-jenkins-csb-rhods.apps.ocp4.prod.psi.redhat.com/job/rhods-ci-pr-test/212/

**Comment by lugi0** [comment]
PR test: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/148/

This turned out to be a bigger PR than I had originally planned, the test has passed (failures are due to expected error outputs in code cells of notebooks).

Since it is quite massive and includes some important refactoring, I would like to see more than one manual review before merging it.

@vasukulkarni @jgarciao @pablofelix @bdattoma @LaVLaS

**Comment by lugi0** [comment]
PR test: https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/rhods-ci-pr-test/160/

**Review by vasukulkarni — APPROVED** [review]

**Comment by vasukulkarni** [comment]
If we go commit by commit, the review is not that hard

**Pull request merged**